### PR TITLE
Separate toasts for existing & new device verification

### PR DESCRIPTION
**Pull request by dbkr**
Screenshot 2020-04-29 at 10 53 27
Screenshot 2020-04-28 at 18 16 47

Separate device verification toasts into ones for devices that were
there when the app loaded and a separate toast for each device that
has appeared since.

Reverts part of https://github.com/matrix-org/matrix-react-sdk/pull/4506
(clicking a device from your own UserInfo now triggers the legacy
verification flow again).

Also converts all the usages of `getStoredDevices()` and `getStoredDevicesForUser ` to synchronous calls now that they're no longer async functions (which, were I more organised, I would have done in a separate PR/commit - sorry).

Fixes https://github.com/vector-im/riot-web/issues/13422
Fixes https://github.com/vector-im/riot-web/issues/13418
Requires https://github.com/matrix-org/matrix-js-sdk/pull/1354

### CODE REVIEW

File: src/DeviceListener.js [hunk 1]
```diff
@@ -79,6 +98,20 @@ export default class DeviceListener {
...
+            const cli = MatrixClientPeg.get();
+            this._ourDeviceIdsAtStart = new Set(
+                cli.getStoredDevicesForUser(cli.getUserId()).map(d => d.deviceId),
+            );
+        }
+    }
+
+    _onWillUpdateDevices = async (users) => {
+        const myUserId = MatrixClientPeg.get().getUserId();
+        if (users.includes(myUserId)) this._ensureDeviceIdsAtStartPopulated();
```
Comment (jryans) [inline src/DeviceListener.js:112]: Should this event version trigger `_recheck`?

File: src/DeviceListener.js [hunk 2]
```diff
@@ -197,32 +232,65 @@ export default class DeviceListener {
...
+            });
+        } else {
+            ToastStore.sharedInstance().dismissToast(OTHER_DEVICES_TOAST_KEY);
+        }
+
+        // Show toasts for new unverified devices if they aren't already there
+        for (const deviceId of newUnverifiedDeviceIds) {
+            ToastStore.sharedInstance().addOrReplaceToast({
+                key: toastKey(deviceId),
+                title: _t("Unverified login. Was this you?"),
```
Comment (jryans) [inline src/DeviceListener.js:280]: [Figma](https://www.figma.com/file/wwqBo5oAFfk8XKmtd2YaIs/Cross-signing?node-id=4592%3A104) says "New login" here.

```suggestion
                title: _t("New login. Was this you?"),
```

File: src/DeviceListener.js [hunk 3]
```diff
@@ -197,32 +232,65 @@ export default class DeviceListener {
...
+                icon: "verification_warning",
+                props: { deviceId },
+                component: sdk.getComponent("toasts.UnverifiedSessionToast"),
```
Comment (jryans) [inline src/DeviceListener.js:283]: Is this component still around? I'm only seeing a move of `UnverifiedSessionToast` to `BulkUnverifiedSessionToast` in this PR, maybe a missing file...?

Comment (dbkr) [reply src/DeviceListener.js:283]: oops, I did indeed miss a file

File: src/DeviceListener.js (hunk 1 above)
Comment (dbkr) [reply src/DeviceListener.js:112]: Nope - have added a comment to hopefully clarify

**Comment by dbkr** [comment]
Thanks for catching the other `getStoredDevice()` calls - not sure how I missed them

File: src/DeviceListener.js [hunk 4]
```diff
@@ -110,6 +110,9 @@ export default class DeviceListener {
     _onWillUpdateDevices = async (users) => {
         const myUserId = MatrixClientPeg.get().getUserId();
         if (users.includes(myUserId)) this._ensureDeviceIdsAtStartPopulated();
+
+        // No need to do a recheck here: we just need to get a snapshot of our devices
+        // before we download asny new ones.
```
Comment (jryans) [inline src/DeviceListener.js:115]:
```suggestion
        // before we download any new ones.
```

**Pull request merged**